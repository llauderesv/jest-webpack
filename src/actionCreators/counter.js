//  Actions...
export const INC_COUNTER = 'INC_COUNTER';
export const INC_COUNTER_ASYNC = 'INC_COUNTER_ASYNC';
export const DEC_COUNTER = 'DEC_COUNTER';

// Action creators...
export function incrementCounter() {
  return {
    type: INC_COUNTER,
  };
}

export function decrementCounter() {
  return {
    type: DEC_COUNTER,
  };
}

export function incrementAsyncCounter() {
  return { type: INC_COUNTER_ASYNC };
}
