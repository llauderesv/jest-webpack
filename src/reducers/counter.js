import { INC_COUNTER, DEC_COUNTER } from '../actionCreators/counter';

const initialState = { count: 0 };

// Reducer modifying counter state
const reducer = (state = initialState, action) => {
  if (action.type === INC_COUNTER) {
    return { ...state, count: state.count + 1 };
  }

  if (action.type === DEC_COUNTER) {
    return { ...state, count: state.count - 1 };
  }

  return state;
};

export default reducer;
