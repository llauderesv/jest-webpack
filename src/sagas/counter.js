import { put, takeEvery, call } from 'redux-saga/effects';
import { incrementCounter, INC_COUNTER_ASYNC } from '../actionCreators/counter';

// Set a delay utility function...
export const delay = seconds => new Promise(res => setTimeout(res, seconds));

export function* incrementAsync() {
  yield call(delay, 1000);
  yield put(incrementCounter());
}

export default function* watchIncrementAsync() {
  yield takeEvery(INC_COUNTER_ASYNC, incrementAsync);
}
