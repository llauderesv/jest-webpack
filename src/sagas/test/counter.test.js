import 'babel-polyfill';
import { call } from 'redux-saga/effects';
import { incrementAsync, delay } from '../counter';

describe('Counter Saga Tests', () => {
  it('incrementAsync Saga test', () => {
    // This will return a plain objects
    // {done: false, value: (The yielded value)}
    const gen = incrementAsync();

    // Gen next value will return the yielded function/value
    expect(gen.next().value).toMatchSnapshot();
    expect(gen.next().value).toMatchSnapshot();
    // Increment saga must be done...
    expect(gen.next().value).toMatchSnapshot();
  });
});
