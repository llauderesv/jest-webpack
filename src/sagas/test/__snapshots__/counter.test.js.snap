// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Saga Tests incrementAsync Saga test 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      1000,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`Counter Saga Tests incrementAsync Saga test 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "INC_COUNTER",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`Counter Saga Tests incrementAsync Saga test 3`] = `undefined`;
